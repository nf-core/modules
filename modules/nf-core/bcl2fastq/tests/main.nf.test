nextflow_process {

    name "Test Process BCL2FASTQ"
    script "modules/nf-core/bcl2fastq/main.nf"
    process "BCL2FASTQ"
    config "./nextflow.config"
    tag "bcl2fastq"
    tag "modules"
    tag "modules_nfcore"

    test("homo sapiens illumina [bcl]") {
        when {
            process {
                //TODO use new test dataset when available, see https://github.com/nf-core/test-datasets/issues/996
                """
                input[0] = [
                    [ id: 'test', lane:1 ],
                    file(params.test_data['homo_sapiens']['illumina']['test_flowcell_samplesheet'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_flowcell'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
