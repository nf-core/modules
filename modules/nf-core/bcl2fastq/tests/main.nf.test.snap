{
    "homo sapiens illumina [bcl]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        "Sample1_S1_L001_R1_001.fastq.gz:md5,e92fce7b86c6447b053d72c5cb4be89c"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            [
                                [
                                    [
                                        [
                                            [
                                                "lane.html:md5,794e48287f47a9f22dcb6b6d0c22c3eb",
                                                "laneBarcode.html:md5,2bbdae3ee57151eab520c966597d7438"
                                            ],
                                            [
                                                
                                            ]
                                        ]
                                    ],
                                    [
                                        [
                                            [
                                                "lane.html:md5,f741870307050dcea79a838f5971770f",
                                                "laneBarcode.html:md5,ffe2e863811c76cb9da27d5d124e0611"
                                            ],
                                            [
                                                "lane.html:md5,cca97e771d3491dbc8cd3fe389595b09",
                                                "laneBarcode.html:md5,cca97e771d3491dbc8cd3fe389595b09"
                                            ]
                                        ],
                                        [
                                            [
                                                "lane.html:md5,c383b0768d9978733d3f5d3b91c15af0",
                                                "laneBarcode.html:md5,48842c23b9a2816aec540177df870968"
                                            ],
                                            [
                                                
                                            ]
                                        ]
                                    ]
                                ],
                                "Report.css:md5,eb7d3eb68fc1539f411404987246b59b",
                                "index.html:md5,5747c407854ae2c358d0ec201ce622d8",
                                "tree.html:md5,a1b9bf592973ca829ec69ddf888b7e34"
                            ]
                        ]
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "AdapterTrimming.txt:md5,48ed2b914b1246c0b5d8667525550946",
                            "ConversionStats.xml:md5,8fe0f57f3f5d256a0762dba75ac62d05",
                            "DemultiplexingStats.xml:md5,2047ff18f5b9107c084de06e9ff943ad",
                            "DemuxSummaryF1L1.txt:md5,03e5fd0c1e3079c5f8c7b4d0501b37ff",
                            "FastqSummaryF1L1.txt:md5,0c6f2d87ee183b84d1051cde9a5643d1",
                            "Stats.json:md5,8e5f038b8aa9e465599d3575f930e604"
                        ]
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "ControlMetricsOut.bin:md5,6d77b38d0793a6e1ce1e85706e488953",
                            "CorrectedIntMetricsOut.bin:md5,2bbf84d3be72734addaa2fe794711434",
                            "ErrorMetricsOut.bin:md5,38c88def138e9bb832539911affdb286",
                            "ExtractionMetricsOut.bin:md5,7497c3178837eea8f09350b5cd252e99",
                            "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                            "QMetricsOut.bin:md5,7e9f198d53ebdfbb699a5f94cf1ed51c",
                            "TileMetricsOut.bin:md5,83891751ec1c91a425a524b476b6ca3c"
                        ]
                    ]
                ],
                "7": [
                    "versions.yml:md5,12f3dca40b6a65cec6ba51c0fd872152"
                ],
                "fastq": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        "Sample1_S1_L001_R1_001.fastq.gz:md5,e92fce7b86c6447b053d72c5cb4be89c"
                    ]
                ],
                "fastq_idx": [
                    
                ],
                "interop": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "ControlMetricsOut.bin:md5,6d77b38d0793a6e1ce1e85706e488953",
                            "CorrectedIntMetricsOut.bin:md5,2bbf84d3be72734addaa2fe794711434",
                            "ErrorMetricsOut.bin:md5,38c88def138e9bb832539911affdb286",
                            "ExtractionMetricsOut.bin:md5,7497c3178837eea8f09350b5cd252e99",
                            "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                            "QMetricsOut.bin:md5,7e9f198d53ebdfbb699a5f94cf1ed51c",
                            "TileMetricsOut.bin:md5,83891751ec1c91a425a524b476b6ca3c"
                        ]
                    ]
                ],
                "reports": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            [
                                [
                                    [
                                        [
                                            [
                                                "lane.html:md5,794e48287f47a9f22dcb6b6d0c22c3eb",
                                                "laneBarcode.html:md5,2bbdae3ee57151eab520c966597d7438"
                                            ],
                                            [
                                                
                                            ]
                                        ]
                                    ],
                                    [
                                        [
                                            [
                                                "lane.html:md5,f741870307050dcea79a838f5971770f",
                                                "laneBarcode.html:md5,ffe2e863811c76cb9da27d5d124e0611"
                                            ],
                                            [
                                                "lane.html:md5,cca97e771d3491dbc8cd3fe389595b09",
                                                "laneBarcode.html:md5,cca97e771d3491dbc8cd3fe389595b09"
                                            ]
                                        ],
                                        [
                                            [
                                                "lane.html:md5,c383b0768d9978733d3f5d3b91c15af0",
                                                "laneBarcode.html:md5,48842c23b9a2816aec540177df870968"
                                            ],
                                            [
                                                
                                            ]
                                        ]
                                    ]
                                ],
                                "Report.css:md5,eb7d3eb68fc1539f411404987246b59b",
                                "index.html:md5,5747c407854ae2c358d0ec201ce622d8",
                                "tree.html:md5,a1b9bf592973ca829ec69ddf888b7e34"
                            ]
                        ]
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "AdapterTrimming.txt:md5,48ed2b914b1246c0b5d8667525550946",
                            "ConversionStats.xml:md5,8fe0f57f3f5d256a0762dba75ac62d05",
                            "DemultiplexingStats.xml:md5,2047ff18f5b9107c084de06e9ff943ad",
                            "DemuxSummaryF1L1.txt:md5,03e5fd0c1e3079c5f8c7b4d0501b37ff",
                            "FastqSummaryF1L1.txt:md5,0c6f2d87ee183b84d1051cde9a5643d1",
                            "Stats.json:md5,8e5f038b8aa9e465599d3575f930e604"
                        ]
                    ]
                ],
                "undetermined": [
                    
                ],
                "undetermined_idx": [
                    
                ],
                "versions": [
                    "versions.yml:md5,12f3dca40b6a65cec6ba51c0fd872152"
                ]
            }
        ],
        "timestamp": "2023-10-25T11:06:04.763448"
    }
}